recipes/default.rb

```diff
@@ -51,15 +51,26 @@
   action :nothing
 end
 
-template '/etc/init/chef-guard.conf' do
-  source 'chef-guard.upstart.erb'
-  backup false
-  variables(
-    :recipe_file => (__FILE__).to_s.split("cookbooks/")[1],
-    :template_file => source.to_s,
-    :basedir => node['chef-guard']['install_dir']
-  )
-  notifies :restart, 'service[chef-guard]'
+# Check if CentOS/RHEL 7 is used, and thus assume systemd is present
+if node['platform_family'] == 'rhel' && node['platform_version'].to_i >= 7 && node['platform_version'].to_i < 8
```

Comment:
Why are you checking for `>= 7` and `< 8`? I understand the first check, but don't see why we should check for `< 8` it's pretty unlikely they will change `systemd` with the next major release and if it turns out it does, we can always fix it whenever that will be released. But it think changes are far greater that this will just work for feature RHEL versions as well.